toolkit/tools/internal/pkggraph/pkggraph.go

```
@@ -1440,3 +1575,48 @@ func findAllRPMS(rpmsToFind []string) (foundAllRpms bool, missingRpms []string)
 
 	return
 }
+
+// findAllRPMSUpstream returns true if all RPMs requested are found in PMC/repo.
+//  This is used to check if all RPMs are available in PMC/repo before starting a build.
+//  needs repo-list, manifest files rpmrepocloner
+func findAllRPMSUpstream(rpmsToFind []*PkgNode, outDir, tmpDir, workerTar, existingRpmsDir, existingToolchainRpmsDir string, usePreviewRepo, ignoreVersionToResolveSelfDep bool, repoFiles []string) (foundAllRpms bool) {
```

> (foundAllRpms bool) [](http://example.com/codeflow?start=196&length=20)

Please return an error from all functions, which may error out.